testsuite/classlibrary/TestErrorHandling.sc

```diff
@@ -0,0 +1,33 @@
+
+
+TestThrow : UnitTest {
+
+	test_throw_nested_threads {
+
+		fork { // isolate from surrounding thread
```

Comment:
synchronize this test, it shouldn't just spawn a new thread and exit

Reply:
I would have thought he same, but that didn't work for me - it stops the whole test.

Reply:
well you are going to have to make sure the test is complete somehow before leaving the test function

Reply:
It looks to me like this test is using one `next` and one `condition` variable for multiple levels of syncing, which night be invoked simultaneously. I believe you need multiple variables for each, within each Routine's local scope for clarity (possibly excluding the innermost layer). For clarity it would be better to name them differently as well.

There might be a better way to structure this test. I'm really busy for the next week or so but I'm interested in the problem.

Reply:
Also I'm becoming skeptical of manipulating the threads' exceptionHandler directly. As we saw with Pprotect itself, you may be able to simplify the code and avoid bugs by using prTry.

Reply:
I do see your point. In this case, however, I'd like to test exactly the exceptionHandler, so I'd prefer not to use the indirection through prTry.